### PR TITLE
docs: introduction section for admins

### PR DESCRIPTION
**Pull request by pohly**
We have most of the information documented already, but admins may
need some additional guidance on where to start and what's expected of
them. The new section also spells out some dos and don'ts.

Fixes #312, #707

### CODE REVIEW

**Comment by pohly** [comment]
@intelkevinputnam : can you perhaps help review this new documentation section (spelling, grammar, etc.)?

File: docs/install.md [hunk 1]
```diff
@@ -17,6 +17,71 @@
...
+installation can only use either [direct device
+mode](/docs/design.md#direct-device-mode) or [LVM
+device mode](/docs/design.md#direct-device-mode). It is possible to install
+PMEM-CSI twice on the same cluster with different modes, but then:
+
+- The driver names must be different.
+- The installations must run on different nodes by using different node
+  labels, or the "usage" parameter of the LVM mode driver installation
+  one a node must be so that it leaves spaces available for the direct mode
+  driver installation on that same node.
```
Comment (avalluri) [inline docs/install.md:45]: Would it make sense to mention about providing the different `StorageClasses` for those drivers?

Comment (avalluri) [reply docs/install.md:45]: Ignore this, I figured out that the next paragraph is having this.

**Comment by pohly** [comment]
@avalluri so what do you think, is this complete enough? If we don't hear from @intelkevinputnam, shall we merge it?

**Comment by pohly** [comment]
One thing that I noticed is that we don't have good documentation for storage class parameters. `eraseAfter` was only described once, in the context of inline volumes. Probably should be enhanced at some point.

**Comment by pohly** [comment]
Upgrade/downgrade also may deserve a separate section.

**Comment by pohly** [comment]
There is another issue open (https://github.com/intel/pmem-csi/issues/707) which gets (almost) addressed by this PR.

What was missing was this:
> operator in install.md must use the "pmem-csi.intel.com" name to be compatible with the example storage classes

I added another commit for that.

@avalluri: please have another look.

**Comment by avalluri** [comment]
> @avalluri: please have another look.

Looks good.

**Comment by intelkevinputnam** [comment]
@pohly I can take a look this afternoon.

File: docs/install.md [hunk 2]
```diff
@@ -17,6 +17,71 @@
 - [PMEM-CSI Deployment CRD](#pmem-csi-deployment-crd)
 - [Filing issues and contributing](#filing-issues-and-contributing)
 
+## Overview
+
+This section summarizes the steps that may be needed during the entire
```
Comment (intelkevinputnam) [inline docs/install.md:22]: Line 23: replace "usage" with "life cycle".

File: docs/install.md [hunk 3]
```diff
@@ -17,6 +17,71 @@
...
+usage of PMEM in a cluster, starting with the initial preparations and
+ending with decommissioning the hardware. The other sections explain
+each step in more detail.
+
+When setting up a cluster, the administrator must install the PMEM
+hardware on nodes and configure some or all of that PMEM for usage by
```
Comment (intelkevinputnam) [inline docs/install.md:28]: Line 28: possibly replace "that PMEM" with "those instances of PMEM"

File: docs/install.md [hunk 4]
```diff
@@ -17,6 +17,71 @@
...
+PMEM-CSI ([prerequisites](#prerequisites)]. Nodes where PMEM-CSI is
+supposed to run must have a certain [label in
+Kubernetes](#installation-and-setup).
+
+The administrator must install PMEM-CSI, using either [the PMEM-CSI
+operator](#install-using-the-operator) (recommended) or with [scripts
+and YAML files in the source code](#install-from-source). A PMEM-CSI
+installation can only use either [direct device
```
Comment (intelkevinputnam) [inline docs/install.md:36]: Line 36: I'd recommend removing "either"

File: docs/install.md [hunk 5]
```diff
@@ -17,6 +17,71 @@
...
+ending with decommissioning the hardware. The other sections explain
+each step in more detail.
+
+When setting up a cluster, the administrator must install the PMEM
+hardware on nodes and configure some or all of that PMEM for usage by
+PMEM-CSI ([prerequisites](#prerequisites)]. Nodes where PMEM-CSI is
+supposed to run must have a certain [label in
+Kubernetes](#installation-and-setup).
+
+The administrator must install PMEM-CSI, using either [the PMEM-CSI
```
Comment (intelkevinputnam) [inline docs/install.md:33]: Line 33: remove "either"

File: docs/install.md [hunk 6]
```diff
@@ -17,6 +17,71 @@
...
+operator](#install-using-the-operator) (recommended) or with [scripts
+and YAML files in the source code](#install-from-source). A PMEM-CSI
+installation can only use either [direct device
+mode](/docs/design.md#direct-device-mode) or [LVM
+device mode](/docs/design.md#direct-device-mode). It is possible to install
+PMEM-CSI twice on the same cluster with different modes, but then:
```
Comment (intelkevinputnam) [inline docs/install.md:39]: Line 39: recommend replacing "but then" with "with these restrictions"

**Review by intelkevinputnam — CHANGES_REQUESTED** [review]
Please consider these recommendations.

**Comment by pohly** [comment]
@intelkevinputnam : thanks, I've applied all of your suggestions.

**Pull request merged**